src/components/course/LicenseRequestedAlert.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useContext, useMemo } from 'react';
+import { Container, Alert } from '@edx/paragon';
+import Cookies from 'universal-cookie';
+import PropTypes from 'prop-types';
+
+import {
+  LICENSE_REQUESTED_ALERT_DISMISSED_COOKIE_NAME,
+} from './data/constants';
+import { SubsidyRequestsContext } from '../enterprise-subsidy-requests';
+import { SUBSIDY_REQUEST_STATE } from '../enterprise-subsidy-requests/constants';
+import { UserSubsidyContext } from '../enterprise-user-subsidy/UserSubsidy';
+
+/**
+ * A component to render an alert when a learner has a license request that is pending review.
+ * Once dismissed, the learner will not see this alert again until the cookies are cleared.
+ */
+const LicenseRequestedAlert = ({ catalogList }) => {
+  const cookies = new Cookies();
+  const previouslyDismissed = cookies.get(LICENSE_REQUESTED_ALERT_DISMISSED_COOKIE_NAME);
+  const [isAlertOpen, setIsAlertOpen] = useState(!previouslyDismissed);
+
+  const {
+    licenseRequests,
+  } = useContext(SubsidyRequestsContext);
+  const { customerAgreementConfig } = useContext(UserSubsidyContext);
+
+  const pendingLicenseRequest = useMemo(() => licenseRequests.find(
+    request => request.state === SUBSIDY_REQUEST_STATE.REQUESTED,
+  ), [licenseRequests.length]);
+
+  const subscriptionCatalogUUIDs = customerAgreementConfig?.subscriptions?.map(
+    subscription => subscription.enterpriseCatalogUuid,
+  ) ?? [];
+
+  const hasApplicableSubscription = subscriptionCatalogUUIDs.find(uuid => catalogList.includes(uuid));
```

Comment:
Any reason why these variables aren't computed with `useMemo`, like `pendingLicenseRequest`? They all iterate over an array, which could probably be done only when the array changes. 🤔